File: Estruturais/Adapter/Façade/facade01.js
// subsistema 1 
class SistemaPagamento{
    processarPagamento(valor){
        console.log(`Processamento pagamento no valor de R$ ${valor}`);
    }
}

//subsistema 2
class SistemaLogistica{
    enviarProduto(destinatario){
        console.log(`Enviar Produto para: ${destinatario}`);
    }
}

//subsistema 3 
class SistemaNotificacao{
    enviarEmail(destinatario,mensagem){
        console.log(`Enviar e-mail para ${destinatario}: ${mensagem}`);
    }
}

//fachda 
class LojaOnline{
    constructor(){
        this.pagamento = new SistemaPagamento();
        this.Logistica = new SistemaLogistica();
        this.Notificacao = new SistemaNotificacao();
    }

    realizarCompra(produto, destinatario){
        this.pagamento.processarPagamento(produto.preco);
        this.Logistica.enviarProduto(destinatario);
        this.Notificacao.enviarEmail(destinatario, `Seu pedido de ${produto.nome} foi enviado`);

    }
}

//Cliente - exemplo de uso ##########33

const loja = new LojaOnline();

const produto = {
    nome: 'camiseta',
    preco: 49.99
}

const destinatario = 'usuarioteste@teste.com';

loja.realizarCompra(produto,destinatario);

File: Estruturais/Adapter/Façade/facade01.py
# Subsistema 
class SistemaPagamento:
    def processadorPagamento(self, valor):
        print(f"processamento pagamento no valor de R$ {valor}")


#Subsistema 2 
class SistemaLogistica:
    def enviarProduto(self,destinatario):
        print(f"Enviar produto para: {destinatario}")
        
        
#Subsistema 3
class SistemaNotificacao:
    def enviarEmail(self,destinatario, mensagem):
        print(f"Enviar email para: {destinatario}: {mensagem}")
        

#fachada 
class LojaOnline:
    def __init__(self):
        self.pagamento = SistemaPagamento()
        self.logistica = SistemaLogistica()
        self.notificacao = SistemaNotificacao()
        
    def realizarCompra(self,produto,destinatario):
        self.pagamento.processadorPagamento(produto['preco'])
        self.logistica.enviarProduto(destinatario)
        self.notificacao.enviarEmail(destinatario, f"Seu pedido de {produto['nome']} foi enviado")
        
# clinte ]]###

loja = LojaOnline()

produto = {
    "nome": "Camiseta",
    "preco": 49.99
}

destinatario = 'teste@teste.com'

loja.realizarCompra(produto, destinatario)
        
    